### PR TITLE
Fixes #3628 - Disablers no longer blow up mechs

### LINKED ISSUE #3628: Mechs, pods and the like, take damage from tasers and disablers
title

### PR DESCRIPTION
**Pull request by Kyep**
Issue #3628 states that disabler shots can blow up pods and mechs.
After testing, I found that space pods don't take damage from disablers... but mechs do. Mechs also have a message saying they were 'hit' by tasers, even though it obviously does nothing.

This PR fixes #3628 by:
- Making mechs no longer take any damage from disabler shots
- Making taser/disabler shots against mechs now produce the message that the mech's "armour is undamaged" on impact. 

🆑
fix: Disabler shots no longer damage mechs.
fix: On impact, both taser and disabler shots now produce a message saying that the mech is undamaged by them, instead of a message saying they 'hit'.
/🆑

### CODE REVIEW

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -565,6 +565,8 @@
 
 	if(prob(deflect_chance * deflection) && (Proj.damage_type == BRUTE || Proj.damage_type == BURN))
```
Comment (Fox-McCloud) [inline code/game/mecha/mecha.dm]: why not just move the `BRUTE` and `BURN` damage checks out, independently, so you don't have to do specific checks for every single damage type under the sun? There's more than just `STAMINA` that you'd technically have to correct for, here; checking for them all would be messy.

Comment (Kyep) [inline code/game/mecha/mecha.dm]: Not sure what you mean by move them out independently.

Comment (Fox-McCloud) [inline code/game/mecha/mecha.dm]: The brue and burn check should be done separately and before the deflection stuff, that way only projectiles that do brute and burn damage will go through.

**Comment by Kyep** [comment]
@Fox-McCloud Updated

**Comment by taukausanake** [comment]
@Crazylemon64 are you done abusing labels yet?

**Pull request merged**